packages/remirror__extension-entity-reference/__tests__/entity-reference-extension.spec.ts

```diff
@@ -211,4 +211,50 @@ describe('EntityReference marks', () => {
       expect(entityReferences).toHaveLength(2);
     });
   });
+
+  describe('getEntityReferenceId', () => {
+    it('Returns entityReference by Id', () => {
+      add(doc(p('testing text')));
+      const entityReference = {
+        id: 'testId',
+        from: 1,
+        to: 8,
+        text: 'testing',
+      };
+      selectText({ from: entityReference.from, to: entityReference.to });
+      commands.addEntityReference(entityReference.id);
+      const entityReferenceFromDoc = helpers.getEntityReferenceById(entityReference.id);
+
+      expect(entityReferenceFromDoc).toEqual(entityReference);
+    });
+  });
+
+  describe('scrollIntoEntityReference', () => {
+    it('returns correct range when the selection is not empty', () => {
+      const editor = add(doc(p(DUMMY_TEXT)));
+      const entityReference = {
+        id: 'testId',
+        from: 3,
+        to: 8,
+        text: 'testing',
+      };
+      commands.selectText({ from: entityReference.from, to: entityReference.to });
+      commands.addEntityReference(entityReference.id);
+
+      // Fire an update to remove the current range selection.
+      // The cursor will be placed at the anchor of the current range selection.
+      // in this case the anchor is 3.
+      editor.commands.emptySelection();
```

Comment:
Is worth setting the selection to somewhere arbitrary in the doc instead? So we can check the from position is updated properly too?

Reply:
Done ✅ , I selected all the document instead.